libres/tests/analysis/test_update.cpp

```diff
@@ -0,0 +1,341 @@
+        double b_ml = obs_mean - a_ml * iobs_mean;
+
+        matrix_type *E = obs_data_allocE(obs_data, rng, ens_size);
+
+        int nparam = 2;
+        matrix_type *A = matrix_alloc(nparam, ens_size);
+        for (int iens = 0; iens < ens_size; iens++) {
+            const auto &model = ens[iens];
+            matrix_iset(A, 0, iens, model.a);
+            matrix_iset(A, 1, iens, model.b);
+        }
+        double a_avg_prior = matrix_get_row_sum(A, 0) / ens_size;
+        double b_avg_prior = matrix_get_row_sum(A, 1) / ens_size;
+
+        matrix_type *C_psi_f = covariance_matrix(A);
+        double det_C_psi_f = matrix_det(C_psi_f);
+
+        // std::unordered_map<std::string, matrix_type *> parameters = {{"A", A}};
+
+        WHEN("Running update without rowscaling") {
+            analysis::run_analysis_update_without_rowscaling(
+                enkf_module, ens_mask, meas_data, obs_data, rng, E, A);
+
+            double a_avg_posterior = matrix_get_row_sum(A, 0) / ens_size;
+            double b_avg_posterior = matrix_get_row_sum(A, 1) / ens_size;
+
+            double d_prior_ml = std::sqrt(std::pow((a_avg_prior - a_ml), 2) +
+                                          std::pow((b_avg_prior - b_ml), 2));
+            double d_posterior_ml =
+                std::sqrt(std::pow((a_avg_posterior - a_ml), 2) +
+                          std::pow((b_avg_posterior - b_ml), 2));
+
+            double d_prior_posterior =
+                std::sqrt(std::pow((a_avg_prior - a_avg_posterior), 2) +
+                          std::pow((b_avg_prior - b_avg_posterior), 2));
+
+            THEN("Euclidean distances should follow certain rules") {
+                REQUIRE(d_prior_ml > d_posterior_ml);
+                REQUIRE(d_prior_ml > d_prior_posterior);
+            }
+
+            THEN("Parameter estimates improve with increased trust in "
+                 "observations") {
+                // Elementwise distances are stored for one specific ensemble size.
+                // New elements are added to each new obs_std.
+                if (ens_size == 100) {
+                    a_diff_ml_posterior.push_back(
+                        std::pow(a_avg_posterior - a_ml, 2));
+                    b_diff_ml_posterior.push_back(
+                        std::pow(b_avg_posterior - b_ml, 2));
+
+                    if (a_diff_ml_posterior.size() > 1) {
+                        for (int i = 0; i < a_diff_ml_posterior.size() - 1;
+                             i++) {
+                            REQUIRE(a_diff_ml_posterior[i] >
+                                    a_diff_ml_posterior[i + 1]);
+                            REQUIRE(b_diff_ml_posterior[i] >
+                                    b_diff_ml_posterior[i + 1]);
+                        }
+                    }
+                }
+            }
+
+            THEN("Parameter estimates converge to ML-estimates") {
+                if (obs_std == 0.00001) {
```

Comment:
Could we require something if `obs_std != 0.0001`?

Reply:
Indeed, thanks.

Reply:
Perhaps, but the point of this test is to make sure that the covariance determinant converges to zero as the errors in observations approach zero, and `obs_std=0.00001` is supposed to simulate `obs_std` approaching zero.